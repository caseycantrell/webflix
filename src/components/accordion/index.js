import { useState } from "react";

export default function Accordion({ children, ...restProps }){
    return (
        <Container {...restProps}>
            <Inner>
                {children}
            </Inner>
        </Container>
    )
};

Accordion.Title = function AccordionTitle({ children, ...restProps }) {
    return <Title {...restProps}></Title>
};

Accordion.Frame = function AccordionFrame({ children, ...restProps }) {
    return <Frame {...restProps}></Frame>
};

Accordion.Item = function AccordionItem({ children, ...restProps }) {
    const [toggleShow, setToggleShow] = useState(false);
    return <Item {...restProps}></Item>
};

Accordion.Header = function AccordionHeader({ children, ...restProps }) {
    return <Header {...restProps}></Header>
};