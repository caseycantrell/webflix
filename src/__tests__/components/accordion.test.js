import { render, fireEvent } from "@testing-library/react";
import faqsData from "../../fixtures/faqs.json";
import { Accordion } from "../../components"; 

describe('<Accordion />', () => {
    it('renders the <Accordion /> with tasty data', () => {
      const { container, getByText } = render(
        <Accordion>
          <Accordion.Title>Frequently Asked Questions</Accordion.Title>
          <Accordion.Frame>
            {faqsData.map((item) => (
              <Accordion.Item key={item.id}>
                <Accordion.Header>{item.header}</Accordion.Header>
                <Accordion.Body>{item.body}</Accordion.Body>
              </Accordion.Item>
            ))}
          </Accordion.Frame>
        </Accordion>
      );

        expect(getByText("Frequently Asked Questions")).toBeTruthy();
        expect(getByText("What is Webflix?")).toBeTruthy();
        expect(getByText("How much does Webflix cost?")).toBeTruthy();
        expect(getByText("Where can I watch?")).toBeTruthy();
        expect(getByText("How do I cancel?")).toBeTruthy();
        expect(getByText("What can I watch on Webflix?")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();

    });

    it('opens & closes the <Accordion /> component', () => {
        const { container, queryByText } = render(
          <Accordion>
            <Accordion.Title>Frequently Asked Questions</Accordion.Title>
            <Accordion.Frame>
              {faqsData.map((item) => (
                <Accordion.Item key={item.id}>
                  <Accordion.Header>{item.header}</Accordion.Header>
                  <Accordion.Body data-testid="accordion-body">{item.body}</Accordion.Body>
                </Accordion.Item>
              ))}
            </Accordion.Frame>
          </Accordion>
        );

        const whatIsWebflixBodyText = "Webflix is a streaming service that offers a wide variety of award-winning TV shows, films, anime, documentaries and more – on thousands of internet-connected devices. You can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV shows and films are added every week!";

        expect(queryByText(whatIsWebflixBodyText)).toBeFalsy();
        fireEvent.click(queryByText("What is Webflix?"));
        expect(queryByText(whatIsWebflixBodyText)).toBeTruthy();

        fireEvent.click(queryByText("What is Webflix?"));
        expect(queryByText(whatIsWebflixBodyText)).toBeFalsy();
        expect(container.firstChild).toMatchSnapshot();
    });
});
