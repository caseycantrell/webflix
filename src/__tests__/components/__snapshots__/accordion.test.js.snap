// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> opens & closes the <Accordion /> component 1`] = `
<div
  class="sc-iCfMLu dQMKRu"
>
  <div
    class="sc-furwcr bIToec"
  >
    <h1
      class="sc-crHmcD ljpfVO"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="sc-iqseJM bwAKbr"
    >
      <div
        class="sc-pVTFL dQgVsI"
      >
        <div
          class="sc-jrQzAO hKxrEF"
        >
          What is Webflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="sc-pVTFL dQgVsI"
      >
        <div
          class="sc-jrQzAO hKxrEF"
        >
          How much does Webflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="sc-pVTFL dQgVsI"
      >
        <div
          class="sc-jrQzAO hKxrEF"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="sc-pVTFL dQgVsI"
      >
        <div
          class="sc-jrQzAO hKxrEF"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="sc-pVTFL dQgVsI"
      >
        <div
          class="sc-jrQzAO hKxrEF"
        >
          What can I watch on Webflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> renders the <Accordion /> with populated data 1`] = `
<div
  class="sc-iCfMLu dQMKRu"
>
  <div
    class="sc-furwcr bIToec"
  >
    <h1
      class="sc-crHmcD ljpfVO"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="sc-iqseJM bwAKbr"
    >
      <div
        class="sc-pVTFL dQgVsI"
      >
        <div
          class="sc-jrQzAO hKxrEF"
        >
          What is Webflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="sc-pVTFL dQgVsI"
      >
        <div
          class="sc-jrQzAO hKxrEF"
        >
          How much does Webflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="sc-pVTFL dQgVsI"
      >
        <div
          class="sc-jrQzAO hKxrEF"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="sc-pVTFL dQgVsI"
      >
        <div
          class="sc-jrQzAO hKxrEF"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="sc-pVTFL dQgVsI"
      >
        <div
          class="sc-jrQzAO hKxrEF"
        >
          What can I watch on Webflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> renders the <Accordion /> with some sweet, sweet data 1`] = `
<div
  class="sc-iCfMLu dQMKRu"
>
  <div
    class="sc-furwcr bIToec"
  >
    <h1
      class="sc-crHmcD ljpfVO"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="sc-pVTFL dQgVsI"
    >
      <div
        class="sc-jrQzAO hKxrEF"
      >
        What is Webflix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="sc-pVTFL dQgVsI"
    >
      <div
        class="sc-jrQzAO hKxrEF"
      >
        How much does Webflix cost?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="sc-pVTFL dQgVsI"
    >
      <div
        class="sc-jrQzAO hKxrEF"
      >
        Where can I watch?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="sc-pVTFL dQgVsI"
    >
      <div
        class="sc-jrQzAO hKxrEF"
      >
        How do I cancel?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="sc-pVTFL dQgVsI"
    >
      <div
        class="sc-jrQzAO hKxrEF"
      >
        What can I watch on Webflix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> renders the <Accordion /> with tasty data 1`] = `
<div
  class="sc-iCfMLu dQMKRu"
>
  <div
    class="sc-furwcr bIToec"
  >
    <h1
      class="sc-crHmcD ljpfVO"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="sc-iqseJM bwAKbr"
    >
      <div
        class="sc-pVTFL dQgVsI"
      >
        <div
          class="sc-jrQzAO hKxrEF"
        >
          What is Webflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="sc-pVTFL dQgVsI"
      >
        <div
          class="sc-jrQzAO hKxrEF"
        >
          How much does Webflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="sc-pVTFL dQgVsI"
      >
        <div
          class="sc-jrQzAO hKxrEF"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="sc-pVTFL dQgVsI"
      >
        <div
          class="sc-jrQzAO hKxrEF"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="sc-pVTFL dQgVsI"
      >
        <div
          class="sc-jrQzAO hKxrEF"
        >
          What can I watch on Webflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
